File: sdk/scala-sdk-protobuf/src/main/scala/kalix/scalasdk/valueentity/ValueEntityOptions.scala
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.scalasdk.valueentity

import kalix.scalasdk.EntityOptions

/** Root entity options for all value based entities. */
trait ValueEntityOptions extends EntityOptions {
  def withForwardHeaders(headers: Set[String]): ValueEntityOptions
}
object ValueEntityOptions {
  val defaults: ValueEntityOptions = ValueEntityOptionsImpl(Set.empty)

  private[kalix] final case class ValueEntityOptionsImpl(forwardHeaders: Set[String]) extends ValueEntityOptions {

    override def withForwardHeaders(headers: Set[String]): ValueEntityOptions =
      copy(forwardHeaders = headers)

  }
}

File: sdk/java-sdk-protobuf/src/main/java/kalix/javasdk/valueentity/ValueEntityOptions.java
/*
 * Copyright (C) 2021-2024 Lightbend Inc. <https://www.lightbend.com>
 */

package kalix.javasdk.valueentity;

import kalix.javasdk.PassivationStrategy;
import kalix.javasdk.impl.valueentity.ValueEntityOptionsImpl;

import java.util.Set;

/** Root entity options for all value based entities. */
public interface ValueEntityOptions extends kalix.javasdk.EntityOptions {

  @Override
  ValueEntityOptions withForwardHeaders(Set<String> headers);

  /**
   * Create a default entity option for a value based entity.
   *
   * @return the entity option
   */
  static ValueEntityOptions defaults() {
    return ValueEntityOptionsImpl.defaults();
  }
}
